demos/rag_agentic/notebooks/Level2_simple_agentic_with_websearch.ipynb

```diff
@@ -617,6 +620,498 @@
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Prompt chaining with websearch tool and custom location tool\n",
+    "\n",
+    "In this section, we demonstrate a more sophisticated use case that combines location detection with web search. Instead of explicitly providing location information, we'll:\n",
+    "\n",
+    "1. **Automatic Location Detection**: Use the `get_location` tool to automatically determine the user's current location\n",
+    "2. **Contextual Search**: Leverage the detected location to find relevant nearby places\n",
+    "\n",
+    "\n",
+    "For example, when a user asks \"Find coffee shops near me that are open now, have Wi-Fi\", the agent will:\n",
+    "- First detect the user's current location\n",
+    "- Then use that location to search for nearby coffee shops\n",
+    "- Present a comprehensive response with relevant options\n",
+    "\n",
+    "This demonstrates how tools can work together to provide intelligent, context-aware responses without requiring explicit location input from the user."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Requirement already satisfied: geocoder in /opt/anaconda3/envs/stackclientv22/lib/python3.10/site-packages (1.38.1)\n",
+      "Requirement already satisfied: click in /opt/anaconda3/envs/stackclientv22/lib/python3.10/site-packages (from geocoder) (8.1.8)\n",
+      "Requirement already satisfied: future in /opt/anaconda3/envs/stackclientv22/lib/python3.10/site-packages (from geocoder) (1.0.0)\n",
+      "Requirement already satisfied: ratelim in /opt/anaconda3/envs/stackclientv22/lib/python3.10/site-packages (from geocoder) (0.1.6)\n",
+      "Requirement already satisfied: requests in /opt/anaconda3/envs/stackclientv22/lib/python3.10/site-packages (from geocoder) (2.32.3)\n",
+      "Requirement already satisfied: six in /opt/anaconda3/envs/stackclientv22/lib/python3.10/site-packages (from geocoder) (1.17.0)\n",
+      "Requirement already satisfied: decorator in /opt/anaconda3/envs/stackclientv22/lib/python3.10/site-packages (from ratelim->geocoder) (5.1.1)\n",
+      "Requirement already satisfied: charset-normalizer<4,>=2 in /opt/anaconda3/envs/stackclientv22/lib/python3.10/site-packages (from requests->geocoder) (3.3.2)\n",
+      "Requirement already satisfied: idna<4,>=2.5 in /opt/anaconda3/envs/stackclientv22/lib/python3.10/site-packages (from requests->geocoder) (3.7)\n",
+      "Requirement already satisfied: urllib3<3,>=1.21.1 in /opt/anaconda3/envs/stackclientv22/lib/python3.10/site-packages (from requests->geocoder) (2.3.0)\n",
+      "Requirement already satisfied: certifi>=2017.4.17 in /opt/anaconda3/envs/stackclientv22/lib/python3.10/site-packages (from requests->geocoder) (2025.1.31)\n"
+     ]
+    }
+   ],
+   "source": [
+    "!pip install geocoder\n",
```

Comment:
please update the `pyproject.toml` and `uv.lock` to add this dependency instead of installing it in the notebook.   
```suggestion
```